docs/concepts/client/rate-limit.md

```diff
@@ -0,0 +1,41 @@
+(client-post-prefetch)=
+# Rate Limit
+
+There are two ways of applying a rate limit using the {class}`~jina.Client`. 
+1. Set using the `Client` class constructor and defaults to 1000 requests. 
+1. Set the argument when using {meth}`~jina.clients.mixin.PostMixin.post` method. If not provided, the deafult value of
+1000 requests will be used. The method argument will override the argument provided in the `Client` class constructor.
+
+
+The `prefetch` argument controls the number of in flight requests made by the {meth}`~jina.clients.mixin.PostMixin.post` 
+method. Using the default value might overload the {class}`~jina.Flow` especially if the operation characteristics of the `Flow` 
+are unknown. Furthermore the Client can send various types requests which can have varying resource usage in the `Flow`.
+
+For example, a high number of `index` requests can contain a large data payload requiring high input/output operation.
+This increases CPU consumption and eventually lead to a build up of the requests on the Flow. If during 
+the queue of in flight requests is already large, a very light weight `search` request to return the total number of 
+documents in the index might be blocked until the queue of `index` requests can be completely processed. To prevent such a scenario,
```

Comment:
```suggestion
Documents in the index might be blocked until the queue of `index` requests can be completely processed. To prevent such a scenario,
```
Capitalize "Documents" throughout